Makefile

```diff
@@ -83,53 +81,86 @@ install:
 	cp -Ra dbicdh "$(DESTDIR)"/usr/share/openqa/dbicdh
 
 
+.PHONY: check
+check: checkstyle test-with-database
+
 .PHONY: checkstyle
 checkstyle: test-shellcheck
 ifneq ($(CHECKSTYLE),0)
 	PERL5LIB=lib/perlcritic:$$PERL5LIB perlcritic lib
 endif
 
-.PHONY: test
-ifeq ($(TRAVIS),true)
-test: run-tests-within-container
-else
-test: checkstyle test-with-database
-endif
+.PHONY: phony_explicit
+phony_explicit: 
+
+.PHONY: test-with-db
+test-with-db: test-with-db-default
 
-.PHONY: test-unit-and-integration
-test-unit-and-integration:
-	OPENQA_CONFIG= prove ${PROVE_LIB_ARGS} ${PROVE_ARGS}
+test-with-db%: phony_explicit
+	@$(MAKE) $(subst test-with-db,test-with-database,$@)
 
 .PHONY: test-with-database
-test-with-database:
+test-with-database: test-with-database-default
+
+test-with-database%: phony_explicit
 	test -d $(TEST_PG_PATH) && (pg_ctl -D $(TEST_PG_PATH) -s status >&/dev/null || pg_ctl -D $(TEST_PG_PATH) -s start) || ./t/test_postgresql $(TEST_PG_PATH)
-	$(MAKE) test-unit-and-integration TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
+	$(MAKE) $(subst -with-database,,$@) TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
 	-pg_ctl -D $(TEST_PG_PATH) stop
 
-# prepares running the tests within Docker (eg. pulls os-autoinst) and then runs the tests considering
-# the test matrix environment variables
-# note: This is supposed to run within the Docker container unlike `launch-docker-to-run-tests-within`
-#       which launches the container.
-.PHONY: run-tests-within-container
-run-tests-within-container:
-	script/run-tests-within-container
```

Comment:
With removal of this we are loosing the possibility to run tests locally within a container easily, isn't it? I think we want to preserve that.

Reply:
Why container should be different from other environments? You just run the same commands in containers as you do locally, e.g. `make test-with-db`

Reply:
That's not what I wrote. I don't know how to write it any differently, I can only repeat: As far as I remember from discussions we wanted to preserve the easy way to run tests locally within a container so I suggest to not remove that. Comments from other reviewers could help here.

Reply:
I think @andrii-suse meant that the script doesn't actually do anything special. Despite the name it won't run in a container on its own accord

Reply:
yes, sure. Maybe my selection of multiple lines to comment on was a bit misleading. I do not have a problem with changing the Makefile or removing individual targets. What we should preserve is the *general* way to run tests using a container, e.g. also line https://github.com/os-autoinst/openQA/pull/2376/files#diff-b67911656ef5d18c4ae36cb6741b7965L150 and following.